cumulusci/utils/http/multi_request.py

```diff
@@ -1,10 +1,13 @@
+import typing as T
 from concurrent.futures import as_completed
 from itertools import chain
-from typing import Dict, Iterable
 
+from requests.exceptions import ReadTimeout
 from requests_futures.sessions import FuturesSession
 
-from cumulusci.utils.iterators import iterate_in_chunks
+from cumulusci.utils.iterators import iterate_in_chunks, partition
+
+RECOVERABLE_ERRORS = (ReadTimeout, ConnectionError)
```

Comment:
These are only recoverable if we're doing a non-mutative request like GET. Otherwise we don't know whether the server handled the request or not.